src/transformers/models/mgp_str/configuration_mgp_str.py

```diff
@@ -0,0 +1,144 @@
+}
+
+
+class MGPSTRConfig(PretrainedConfig):
+    r"""
+    This is the configuration class to store the configuration of a [`MGPSTRModel`]. It is used to instantiate an
+    MGPSTR model according to the specified arguments, defining the model architecture. Instantiating a configuration
+    with the defaults will yield a similar configuration to that of the MGPSTR
+    [alibaba-damo/mgp-str-base](https://huggingface.co/alibaba-damo/mgp-str-base) architecture.
+
+    Configuration objects inherit from [`PretrainedConfig`] and can be used to control the model outputs. Read the
+    documentation from [`PretrainedConfig`] for more information.
+
+
+    Args:
+        img_size (`int`, *optional*, defaults to `32x128`):
+            The size (resolution) of each image.
+        patch_size (`int`, *optional*, defaults to `4`):
+            The size (resolution) of each patch.
+        in_chans (`int`, *optional*, defaults to `3`):
+            The number of input channels.
+        max_token_length (`int`, *optional*, defaults to `27`):
+            The max number of output tokens.
+        char_num_classes (`int`, *optional*, defaults to `27`):
+            The number of classes for char head .
+        bpe_num_classes (`int`, *optional*, defaults to `27`):
+            The number of classes for bpe head .
+        wp_num_classes (`int`, *optional*, defaults to `27`):
+            The number of classes for wp head .
+        embed_dim (`int`, *optional*, defaults to 768):
+            The embedding dimension.
+        depth (`int`, *optional*, defaults to 12):
+            Number of hidden layers in the Transformer encoder.
+        num_heads (`int`, *optional*, defaults to 12):
+            Number of attention heads for each attention layer in the Transformer encoder.
+        mlp_ratio (`int`, *optional*, defaults to 12):
+            The ratio of mlp hidden dim to embedding dim.
+        qkv_bias (`bool`, *optional*, defaults to `True`):
+            Whether to add a bias to the queries, keys and values.
+        representation_size (`int`, *optional*, defaults to `None`):
+            Enable and set representation layer (pre-logits) to this value if set.
+        distilled (`bool`, *optional*, defaults to `False`):
+            Model includes a distillation token and head as in DeiT models.
+        drop_rate (`float`, *optional*, defaults to 0.0):
+            The dropout probability for all fully connected layers in the embeddings, encoder.
+        attn_drop_rate (`float`, *optional*, defaults to 0.0):
+            The dropout ratio for the attention probabilities.
+        drop_path_rate (`float`, *optional*, defaults to 0.0):
+            The stochastic depth rate.
+        norm_layer (`str` or `function`, *optional*, defaults to `None`):
+            The normalization layer function (function or string) in the encoder. for example, `"LayerNorm"`.
```

Comment:
We usually don't allow to provide functions as argument in the Transformers library (I know this is done in timm),
so here you can perhaps only support strings